### DISCUSSION TITLE
Copilot doesn't seem to auto-fill entire blocks anymore

### DISCUSSION THREAD

**Discussion by ffxsam** [community/community#9967] [Copilot Conversations]
For some reason, Copilot only auto-fills line by line. So I have to hit enter, wait for it to suggest a line, hit enter again, ad nauseum. It used to fill out entire blocks in one shot.

Did something change, or do I have something set incorrectly?

**Comment by jasonkneen** [comment]
this ^^^ very annoying

**Reply by davecheney** [reply]
@jasonkneen please try https://github.com/orgs/github-community/discussions/9967#discussioncomment-2996703

**Reply by jasonkneen** [reply]
didn't work for me

**Comment by davecheney** [comment]
Please try adding this to `settings.json`
```
    "github.copilot.advanced": {
        "debug.overrideEngine": "copilot-logit",
    },
```
Please report back if this helped.

**Reply by ffxsam** [reply]
Thanks, Dave! Actually, the problem seems to have gone away on its own. But if it pops up again, I'll try this setting.

**Reply by ffxsam** [reply]
@davecheney That setting didn't help. See video:

https://user-images.githubusercontent.com/12532733/175168604-8fe9cdd7-18bc-42ce-90aa-7167269b61df.mp4

**Reply by ffxsam** [reply]
It does work sometimes, if I use a comment to describe what a function should do. But in most cases, it only completes line-by-line like this. Ctrl-enter only shows the one line, as well.

**Reply by davecheney** [reply]
Thank you Can you pls post a screenshot of your copilot settings page.

**Reply by jasonkneen** [reply]
exactly what I got -- I thought it was interference of another extension like TabNine so removed that -- still happens

**Reply by ffxsam** [reply]
@davecheney Sure. Not sure which settings page you mean, so here's both:

<img width="815" alt="CleanShot 2022-06-23 at 10 49 05@2x" src="https://user-images.githubusercontent.com/12532733/175341157-1e27d338-600c-4515-966e-760eb08a4563.png">

VS Code:
```json
  "github.copilot.enable": {
    "*": true,
    "yaml": false,
    "plaintext": false,
    "markdown": false
  },
```

**Reply by psstpakGithub** [reply]
Is this fix definitive?
If this is a copilot bug, that will be fixed, it will be necessary to undo the settings.json editing?
or is not a bug?

**Reply by jasonkneen** [reply]
didn't work for me

**Reply by MulleDK19** [reply]
> Please try adding this to `settings.json`
> 
> ```
>     "github.copilot.advanced": {
>         "debug.overrideEngine": "copilot-logit",
>     },
> ```
> 
> Please report back if this helped.

And how can we fix this in VS2022?..

**Reply by davecheney** [reply]
This workaround shouldn't be needed any more
